### PR TITLE
OPS: Add mac catalyst to GA

### PR DESCRIPTION
**Pull request by marcosrdz**
## Summary by CodeRabbit

- **New Features**
	- Introduced support for Mac Catalyst, allowing the app to run on macOS.
	- Updated workflow to include separate build steps and TestFlight uploads for Mac Catalyst.
	- Added dedicated Fastlane lanes for building and uploading Mac Catalyst applications.

- **Enhancements**
	- Improved clarity and maintainability of application identifier management.
	- Enhanced upload process to TestFlight with specific naming conventions for both platforms.
	- Streamlined build workflow with distinct job structures for iOS and Mac Catalyst.
	- Refined handling of provisioning profiles for smoother deployment processes.

### CODE REVIEW

File: .github/workflows/build-ios-release-pullrequest.yml [hunk 1]
```diff
@@ -81,6 +81,17 @@ jobs:
         run: |
           bundle exec fastlane ios setup_provisioning_profiles
         working-directory: ./ios
+      - name: Setup Provisioning Profiles (Mac Catalyst)
+        env:
+          MATCH_PASSWORD: ${{ secrets.MATCH_PASSWORD }}
```
Comment (Overtorment) [inline .github/workflows/build-ios-release-pullrequest.yml:86]: whats match password?

File: .github/workflows/build-ios-release-pullrequest.yml [hunk 2]
```diff
@@ -122,14 +133,22 @@ jobs:
...
+        working-directory: ./ios
+      - name: Upload IPA as Artifact (iOS)
         uses: actions/upload-artifact@v2
         with:
           name: BlueWallet.${{env.PROJECT_VERSION}}(${{env.NEW_BUILD_NUMBER}}).ipa
           path: ./ios/build/BlueWallet.${{env.PROJECT_VERSION}}(${{env.NEW_BUILD_NUMBER}}).ipa
+      - name: Upload Mac Catalyst App as Artifact
+        uses: actions/upload-artifact@v2
+        with:
+          name: BlueWallet-Catalyst.${{env.PROJECT_VERSION}}(${{env.NEW_BUILD_NUMBER}}).app
```
Comment (Overtorment) [inline .github/workflows/build-ios-release-pullrequest.yml:150]: whats PROJECT_VERSION ? did you add it to project env vars as well?

**Review by Overtorment — CHANGES_REQUESTED** [review]
title and description lack explanation whats going on here (rabbit does not count)

File: .github/workflows/build-ios-release-pullrequest.yml (hunk 1 above)
Comment (marcosrdz) [reply .github/workflows/build-ios-release-pullrequest.yml:86]: Fastlane repo

File: .github/workflows/build-ios-release-pullrequest.yml (hunk 2 above)
Comment (marcosrdz) [reply .github/workflows/build-ios-release-pullrequest.yml:150]: Its generated with timestamp

File: .github/workflows/build-ios-release-pullrequest.yml (hunk 1 above)
Comment (Overtorment) [reply .github/workflows/build-ios-release-pullrequest.yml:85]: whats MATCH  ??

**Pull request closed without merging**